pilot/pkg/model/push_context.go

```diff
@@ -1038,6 +1042,9 @@ func (ps *PushContext) initServiceRegistry(env *Environment) error {
 		}
 		ps.ServiceByHostnameAndNamespace[s.Hostname][s.Attributes.Namespace] = s
 		ps.ServiceByHostname[s.Hostname] = s
+		s.Mutex.RLock()
+		ps.ClusterVIPs[s] = s.ClusterVIPs
```

Comment:
This can cause panic , the map is still that map

Reply:
Ah. Thanks. Fixed